primitives/src/json.rs

```diff
@@ -0,0 +1,201 @@
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use serde_json::Value;
+
+#[allow(missing_docs)]
+#[derive(Debug, Clone, Eq, Encode, Decode)]
+pub struct JsonString(pub String);
+
+/// A candidate for a valid `JsonString`.
+/// Is used for generalizing ser/de any types to `JsonString` and vise versa
+#[derive(Serialize, Deserialize, Clone)]
+struct JsonCandidate<T>(T);
+
+impl JsonString {
+    /// Construct [`JsonString`]
+    pub fn new<T: Serialize>(payload: T) -> Self {
+        JsonCandidate(payload).try_into().unwrap()
+    }
+
+    /// Tries cast [`JsonString`] to any value.
+    ///
+    /// # Errors
+    /// - if String == invalid Json type
+    pub fn try_into_any<T: DeserializeOwned>(&self) -> Result<T, serde_json::Error> {
+        serde_json::from_str(&self.0)
+    }
+}
+impl PartialEq for JsonString {
+    fn eq(&self, other: &Self) -> bool {
+        serde_json::from_str::<Value>(&self.0).unwrap()
+            == serde_json::from_str::<Value>(&other.0).unwrap()
+    }
+}
+
+impl<'de> serde::de::Deserialize<'de> for JsonString {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let json = Value::deserialize(deserializer)?;
+        Ok(Self(json.to_string()))
+    }
+}
+
+impl serde::ser::Serialize for JsonString {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let json: Value = serde_json::from_str(&self.0).map_err(serde::ser::Error::custom)?;
+        json.serialize(serializer)
+    }
+}
+
+impl iroha_schema::TypeId for JsonString {
+    fn id() -> iroha_schema::Ident {
+        "JsonString".to_owned()
+    }
+}
+
+impl IntoSchema for JsonString {
+    fn type_name() -> iroha_schema::Ident {
+        <Self as iroha_schema::TypeId>::id()
+    }
+
+    fn update_schema_map(map: &mut iroha_schema::MetaMap) {
+        if !map.contains_key::<Self>() {
+            map.insert::<Self>(iroha_schema::Metadata::String);
+        }
+    }
+}
+
+impl<T: Serialize> TryFrom<JsonCandidate<T>> for JsonString {
+    type Error = serde_json::Error;
+    fn try_from(value: JsonCandidate<T>) -> Result<Self, Self::Error> {
+        Ok(JsonString(serde_json::to_string(&value.0)?))
+    }
+}
+
+impl PartialOrd for JsonString {
+    fn partial_cmp(&self, other: &Self) -> Option<core::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for JsonString {
+    fn cmp(&self, other: &Self) -> core::cmp::Ordering {
+        let l: Value = serde_json::from_str(&self.0).unwrap();
+        let r: Value = serde_json::from_str(&other.0).unwrap();
+
+        match (l, r) {
+            (Value::Number(left), Value::Number(right)) => {
+                if left.is_u64() && right.is_u64() {
+                    left.as_u64().cmp(&right.as_u64())
+                } else {
+                    left.as_f64().unwrap().total_cmp(&right.as_f64().unwrap())
+                }
```

Comment:
I'm afraid this part might panic where we don't want to.

Reply:
I can add if is_f64 check. Then there should be no panic variants